aten/src/ATen/native/cuda/Copy.cu

```diff
@@ -302,9 +302,11 @@ static void copy_kernel_cuda(TensorIterator& iter, bool non_blocking) {
     Tensor src_contig;
 
     // If non_blocking is true - type conversions are performed on the GPU
-    // for CPU-GPU copies, otherwise type conversions are performed on the CPU.
```

Comment:
I'm confused by this comment. For GPU -> CPU copies with non_blocking, the old code was already doing to dtype conversion on the src device which is the GPU, not the CPU?

Reply:
Yeah the comment was wrong (or they have a different meaning for `CPU-GPU copy`, after all it doesn't specify if it's `CPU-to-GPU` or `to-CPU-from-GPU`. Implied prepositions are important!